### PR TITLE
fix: various cleanups for release-tag action

### PR DESCRIPTION
**Pull request by faiq**
**What problem does this PR solve?**:
A couple of changes here

1. Just use the tag name -> https://github.community/t/how-to-get-just-the-tag-name/16241/35
2. Have the checksum file include the `v` in the file name
3. Add a `needs` for the notarizing pipeline 
4. uses new checksumfile 

**Which issue(s) does this PR fix?**:

* https://jira.d2iq.com/browse/D2IQ-NUMBER

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: .github/workflows/release-tag.yaml [hunk 1]
```diff
@@ -57,11 +57,12 @@ jobs:
...
           add-paths: preprovisioned/kib
           body: bumps KIB to latest version
-          commit-message: "fix: bump kib to ${{ github.ref }}"
+          commit-message: "fix: bump kib to ${{ github.ref_name }}"
           path: cluster-api-provider-preprovisioned
-          title: "fix: bump kib to ${{ github.ref }}"
+          title: "fix: bump kib to ${{ github.ref_name }}"
 
   sign-binary:
+    needs: release-artifacts
```
Comment (faiq) [inline .github/workflows/release-tag.yaml:65]:
```suggestion
    needs: upload-release-artifacts
```

Comment (dkoshkin) [reply .github/workflows/release-tag.yaml:65]: The job name is `release-artifacts`, you wouldn't want the suggested change.

**Review by dkoshkin — APPROVED** [review]

File: .github/workflows/release-tag.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@ jobs:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0
-          ref: ${{ github.ref }}
+          ref: ${{ github.ref_name }}
```
Comment (dkoshkin) [inline .github/workflows/release-tag.yaml:14]: The previous run worked fine here https://github.com/mesosphere/konvoy-image-builder/runs/7437609688?check_suite_focus=true
```suggestion
          ref: ${{ github.ref}}
```

**Review by dkoshkin — CHANGES_REQUESTED** [review]

File: .github/workflows/release-tag.yaml (hunk 2 above)
Comment (dkoshkin) [reply .github/workflows/release-tag.yaml:14]: The `ref` here most likely needs to full ref not just the name.

**Review by dkoshkin — APPROVED** [review]

File: .github/workflows/release-tag.yaml [hunk 3]
```diff
@@ -11,7 +11,7 @@ jobs:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0
-          ref: ${{ github.ref }}
+          ref: ${{ github.ref}}
```
Comment (dkoshkin) [inline .github/workflows/release-tag.yaml:14]:
```suggestion
          ref: ${{ github.ref }}
```

Comment (faiq) [reply .github/workflows/release-tag.yaml]: thank you

**Comment by dkoshkin** [comment]
Lets try it again.

**Review by supershal — APPROVED** [review]

**Pull request merged**